### PR TITLE
[TechDraw] Refactor cosmetic.cpp

### PR DESCRIPTION
**Pull request by benj5378**
### [TechDraw] Refactor cosmetic.cpp

- Gather rotation calculations in a single function
- Use default parameters for LineFormat constructor
- Use constructor delegation and default parameters for CosmeticEdge
- Refactor CosmeticEdge constructors
- Improve readability of Cosmetic.cpp
- Gather rotation calculations in a single function

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [x]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

File: src/Mod/TechDraw/App/Cosmetic.cpp [hunk 1]
```diff
@@ -280,20 +272,14 @@ boost::uuids::uuid CosmeticVertex::getTag() const
...
 {
     // Initialize a random number generator, to avoid Valgrind false positives.
     static boost::mt19937 ran;
-    static bool seeded = false;
-
-    if (!seeded) {
-        ran.seed(static_cast<unsigned int>(std::time(nullptr)));
-        seeded = true;
-    }
+    ran.seed(static_cast<unsigned int>(std::time(nullptr)));
```
Comment (benj5378) [inline src/Mod/TechDraw/App/Cosmetic.cpp:282]: @WandererFan Can you please make sure that this doesn't do any harm?

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7194">pipeline status</a> for feature branch [PR_7194](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7194). Pipeline [589479493](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/589479493) was triggered at [9188935](https://github.com/FreeCAD/FreeCAD/pull/7194/commits/91889352ee79b2fb0af54373850efd1e0c2dd203). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/TechDraw/App/Cosmetic.cpp (hunk 1 above)
Comment (WandererFan) [reply src/Mod/TechDraw/App/Cosmetic.cpp:282]: This doesn't do the same thing as the replaced code.  seeded is static so ran.seed only gets called the first time through.   The new code resets the random number generator every time.

This same code is in Part and PartDesign, so I would leave it as is.

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7194">pipeline status</a> for feature branch [PR_7194](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7194). Pipeline [589536284](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/589536284) was triggered at [1e30dc6](https://github.com/FreeCAD/FreeCAD/pull/7194/commits/1e30dc62c744c5cb096b080d279a0518e8c290f9). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7194">pipeline status</a> for feature branch [PR_7194](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7194). Pipeline [589536968](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/589536968) was triggered at [11d36e7](https://github.com/FreeCAD/FreeCAD/pull/7194/commits/11d36e73afe7185e5ed6e84ad3be78fbf89a5915). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by donovaly** [comment]
This PR made master for me uncompilable: https://forum.freecadweb.org/viewtopic.php?f=10&t=70316

**Comment by benj5378** [comment]
> This PR made master for me uncompilable: https://forum.freecadweb.org/viewtopic.php?f=10&t=70316

@donovaly I tried recompiling to this commit https://github.com/FreeCAD/FreeCAD/pull/7194/commits/11d36e73afe7185e5ed6e84ad3be78fbf89a5915 without any issues.

```
[ 97%] Building CXX object src/Mod/TechDraw/Gui/CMakeFiles/TechDrawGui.dir/ViewProviderRichAnno.cpp.o
[ 97%] Building CXX object src/Mod/TechDraw/Gui/CMakeFiles/TechDrawGui.dir/ViewProviderWeld.cpp.o
[ 97%] Linking CXX shared library ../../../../Mod/TechDraw/TechDrawGui.so
[100%] Built target TechDrawGui
benj5378@benj5378-Z68A-D3H-B3:~/development/fc/freecad-build2$
```

I compiled on Ubuntu with regards to the guide on the Wiki. This was not a clean build however. As far as I could see on the forum, you're on Windows - wondering if that had anything to do with it.

Screenshot from 2022-07-17 01-05-35

I am suspecting if the code order could have anything to do with it? Could you try moving this code up before their reference and try compiling again?

**Comment by donovaly** [comment]
> I am suspecting if the code order could have anything to do with it?


Yes, this was the bug. Code was accessed that was declared later. I wonder why this compiles for Clang.

I have already fixed it: a4ec6d4f83

**Pull request merged**